### DISCUSSION TITLE
No easy way to send a reply to caller actor ?

### DISCUSSION THREAD

**Discussion by coldwarrl** [slawlor/ractor#176] [Q&A] [answered]
Hi,

in akka, I can just use the handle sender to send a reply message like
`sender.Tell(my_reply)`

Do I have to implement my own protocol (e.g. add result actor field in message struct) to be able to send a response to the calling actor ? 

Thanks!

**Comment by slawlor** [comment, marked as answer]
You can see an example in https://slawlor.github.io/ractor/quickstart/#adding-state

But the short answer is your message type needs to include the reply channel which is an `RpcReplyPort<_>` and the actor uses that to send the reply back.

The caller just needs to use either the `call!` macro or `ActorRef<_>.call` which will handle sending the message and waiting for the reply.

**Comment by coldwarrl** [comment]
Thanks! I have overseen that.

Is this the Erlang way to do this, or is there any reason not include the sender like in Akka ? My experience with actors are solely from Akka/Akka.net...

**Reply by slawlor** [reply]
In Erlang they would send the actor along with a reference to the specific message. This is because the caller is likely blocking on the reply, so the runtime needs to match up the sent message with the received reply. We do this by using a channel, which is semantically the same, and if you wanted you could construct the channel and not wait on a reply right away if you were doing say other work while that processing was occurring.

You also could send an actor ref instead of a channel and your downstream actor just sends a blind message back to that actor, which sounds like what it's doing in akka. The problem is then you have to match up the message to the reply manually which can be annoying sometimes
